source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled.cs

```diff
@@ -504,19 +504,29 @@ public async Task DuringCompleteScript_ScriptExecutionCanBeCancelled(TentacleCon
             scriptServiceV2CallCounts.CompleteScriptCallCountStarted.Should().BeGreaterOrEqualTo(1);
         }
 
-        private void PauseOrStopPortForwarder(RpcCallStage rpcCallStage, PortForwarder portForwarder, IResponseMessageTcpKiller responseMessageTcpKiller, Reference<bool> rpcCallHasStarted)
+        private async Task PauseOrStopPortForwarder(RpcCallStage rpcCallStage, PortForwarder portForwarder, IResponseMessageTcpKiller responseMessageTcpKiller, Reference<bool> rpcCallHasStarted)
         {
             if (rpcCallStage == RpcCallStage.Connecting)
             {
                 Logger.Information("Killing the port forwarder so the next RPCs are in the connecting state when being cancelled");
                 //portForwarder.Stop();
                 portForwarder.EnterKillNewAndExistingConnectionsMode();
-                rpcCallHasStarted.Value = true;
+
+                await SetRpcCallWeAreInterestedInAsStarted(rpcCallHasStarted);
             }
             else
             {
                 Logger.Information("Will Pause the port forwarder on next response so the next RPC is in-flight when being cancelled");
-                responseMessageTcpKiller.PauseConnectionOnNextResponse(() => rpcCallHasStarted.Value = true);
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+                responseMessageTcpKiller.PauseConnectionOnNextResponse(() => SetRpcCallWeAreInterestedInAsStarted(rpcCallHasStarted));
+#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            }
+
+            static async Task SetRpcCallWeAreInterestedInAsStarted(Reference<bool> rpcCallHasStarted)
+            {
+                // Allow the port forwarder some time to stop
+                await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
I know you would have done this if it were possible, but is there a way we can detect that the port forwarder has actually stopped?

Reply:
I didn't look into it. It would be great to be able to know precisely when everything is stopped but it also sounds like a possible time sync and we are at the mercy of the Operating System and how it is actually dealing with TCP Connections. 

I'm sure this area could be dramatically improved, and all these nasty, prone to flake or fragile with change tests could be improved.

At the moment I'm going for the simplest (unless it's just horrible) approach to try and squish the flakey tests we have!!